dace/codegen/dispatcher.py

```diff
@@ -143,17 +143,17 @@ class TargetDispatcher(object):
     def __init__(self, framecode):
         # Avoid import loop
         from dace.codegen.targets import framecode as fc
+        from dace.codegen import instrumentation
 
         self.frame: fc.DaCeCodeGenerator = framecode
         self._used_targets: Set[target.TargetCodeGenerator] = set()
         self._used_environments = set()
 
-        # type: Dict[dace.dtypes.InstrumentationType, InstrumentationProvider]
-        self.instrumentation = {}
+        self.instrumentation: Dict[Union[dtypes.InstrumentationType, dtypes.DataInstrumentationType],
```

Comment:
Hm, I'm not a huge fan of having to do these unions with two so semantically and syntactically similar types. Semantiaclly, wouldn't it make sense to also include the `Save` and `Restore` from `DataInstrumentation` as e.g. `Instrumentation.Data_Save` and `Instrumentation.Data_Restore`? What would be an argument against that? That these operations would only have an effect on AccessNodes and not e.g. Tasklets, if chosen?

Reply:
The argument is that you may select the wrong kind of instrumentation on access nodes or tasklets. I prefer the comboboxes to be sanitized.

Reply:
Ok I see, that makes sense.